src/library_glfw.js

```diff
@@ -1000,12 +1000,48 @@ var LibraryGLFW = {
       }
     },
 
+  createWebGLContext: function(/** @type {HTMLCanvasElement} */ canvas, webGLContextAttributes) {
+      var ctx;
+      var contextHandle;
+
+        // For GLES2/desktop GL compatibility, adjust a few defaults to be different to WebGL defaults, so that they align better with the desktop defaults.
+        var contextAttributes = {
+          antialias: false,
+          alpha: false,
+#if MIN_WEBGL_VERSION >= 2
+          majorVersion: 2,
+#elif MAX_WEBGL_VERSION >= 2 // library_browser.js defaults: use the WebGL version chosen at compile time (unless overridden below)
+          majorVersion: (typeof WebGL2RenderingContext != 'undefined') ? 2 : 1,
+#else
+          majorVersion: 1,
+#endif
+        };
+
+        if (webGLContextAttributes) {
+          for (var attribute in webGLContextAttributes) {
+            contextAttributes[attribute] = webGLContextAttributes[attribute];
+          }
+        }
+
+        // This check of existence of GL is here to satisfy Closure compiler, which yells if variable GL is referenced below but GL object is not
+        // actually compiled in because application is not doing any GL operations. TODO: Ideally if GL is not being used, this function
+        // Browser.createContext() should not even be emitted.
+        if (typeof GL != 'undefined') {
+          contextHandle = GL.createContext(canvas, contextAttributes);
+          if (contextHandle) {
+            ctx = GL.getContext(contextHandle).GLctx;
+          }
+        }
+
+      if (!ctx) return null;
+
+      return contextHandle;
```

Comment:
Why not use existing `emscripten_webgl_create_context` function for this?

Reply:
because it returns a context, not a handle. is there a way to get the handle from an existing context?

Reply:
`emscripten_webgl_do_create_context` (which is used to implement `emscripten_webgl_create_context`) looks like it returns context handle, no?

https://github.com/emscripten-core/emscripten/blob/67ebee3261629f7e3c2bd24b61098af0c730d8d9/src/library_html5_webgl.js#L206-L207

(I'm not very familiar with code so I could be wrong)

Reply:
Looks good! Gonna try. Thx!